_api-reference/index-apis/flush.md

```diff
@@ -0,0 +1,76 @@
+---
+layout: default
+title: Flush
+parent: Index APIs
+nav_order: 36
+---
+
+# Flush
+
+**Introduced 1.0**
+{: .label .label-purple }
+
+The Flush API manually flushes in-memory data from an index onto disk. This ensures that any memory from the index stored in the buffer persists on disk, as opposed to writing all buffer data to the disk, which can affect the performance of your cluster.
```

Comment:
The Flush API permanently stores all the in-memory operation to the lucene index on disk. OpenSearch automatically performs the flush in background based on certain conditions such as the size of transaction log (controlled by `index.translog.flush_threshold_size`) vs performing for each operation for performance reasons. The operations that are flushed to lucene index are not needed to be maintained in the transaction log anymore to recover in the event of the process restart.

Reply:
@sohami: Does whether or not the indexing occured with Lucene matter in the cast of this API? In other words, is mentioning Lucene in the API definition required?

Reply:
All the on-disk representation of the data is lucene index, hence the mention. Or we can say `... in-memory operation to the segments on disk. The operations that are flushed to the segments are not required....`. Also tagging @mgodwan from indexing team to take a second look at the document